analyzers/tests/SonarAnalyzer.UnitTest/Wrappers/ObjectCreationFactoryTest.cs

```diff
@@ -0,0 +1,100 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.UnitTest.MetadataReferences;
+using SonarAnalyzer.Wrappers;
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.UnitTest.Wrappers
+{
+    [TestClass]
+    public class ObjectCreationFactoryTest
+    {
+        [TestMethod]
+        public void ObjectCreationSyntax()
+        {
+            var compilation = CreateCompilation("public class A{public int X;public A(int y){}} public class B{void Foo(){var bar =new A(1){X = 2};}}");
```

Comment:
There's quite a lot of stuff going on there. Because we have two classes, I'd use verbatim multiline string instead (with as few new lines as possible)
```
  var code = @"
public class A
{
    public int X;
    public A(int y) { }
}
public class B
{
    void Foo()
    {
        var bar = new A(1) { X = 2 };
    }
}";
```